core/Archive.php

```diff
@@ -881,26 +891,26 @@ public static function getPluginForReport($report)
     }
 
     /**
-     * @param $archiveGroups
+     * @param $archiveNamesByPlugin
      * @param $site
      * @param $period
      */
-    private function prepareArchive(array $archiveGroups, Site $site, Period $period)
+    private function prepareArchive(array $archiveNamesByPlugin, Site $site, Period $period)
     {
         $coreAdminHomeApi = API::getInstance();
 
-        $requestedReport = null;
-        if (SettingsServer::isArchivePhpTriggered()) {
-            $requestedReport = Common::getRequestVar('requestedReport', '', 'string');
-        }
+        $requestedReport = $this->getRequestedReport();
+
+        $shouldOnlyProcessRequestedArchives = empty($requestedReport)
```

Comment:
In the `private function getDoneStringForPlugin($plugin, $idSites)` method this variable is called `$shouldOnlyProcessRequestedRecords` with a very similar logic deriving the value. Should it be named the same, does it have the same purpose/use?

Reply:
I don't understand the question, should what be named the same? The variable `$shouldOnlyProcessRequestedRecords`? The method `getDoneStringForPlugin()`?

Reply:
Sorry, I need to try and make my questions a bit more explicit, sometimes they get too convoluted.

The question was whether the variable here (`$shouldOnlyProcessRequestedArchives`) should be named `$shouldOnlyProcessRequestedRecords`, similar to the name in the `getDoneStringForPlugin` method, or there's a good reason for them to be named differently, even though the logic of how they get their value is nearly the same.

Reply:
It can be named the same.